dbt/adapters/databricks/python_submissions.py

```diff
@@ -231,6 +241,59 @@ def destroy(self, context_id: str) -> str:
             )
         return response.json()["id"]
 
+    def get_cluster_status(self) -> Dict:
+
+        # https://docs.databricks.com/dev-tools/api/latest/clusters.html#get
+
+        response = requests.get(
+            f"https://{self.host}/api/2.0/clusters/get",
+            headers=self.auth_header,
+            json={"cluster_id": self.cluster_id},
+        )
+        if response.status_code != 200:
+            raise dbt.exceptions.DbtRuntimeError(
+                f"Error getting status of cluster.\n {response.content!r}"
+            )
+
+        json_response = response.json()
+        return json_response
+
+    def start_cluster(self) -> None:
+        """Send the start command and poll for the cluster status until it shows "Running"
+
+        Raise an exception if the restart exceeds our timeout.
+        """
+
+        # https://docs.databricks.com/dev-tools/api/latest/clusters.html#start
+
+        logger.debug(f"Sending restart command for cluster id {self.cluster_id}")
+
+        response = requests.post(
+            f"https://{self.host}/api/2.0/clusters/start",
+            headers=self.auth_header,
+            json={"cluster_id": self.cluster_id},
+        )
+        if response.status_code != 200:
+            raise dbt.exceptions.DbtRuntimeError(
+                f"Error starting terminated cluster.\n {response.content!r}"
+            )
+
+        # seconds
+        MAX_CLUSTER_START_TIME = 900
+        start_time = time.time()
+
+        def get_elapsed() -> float:
+            return time.time() - start_time
+
+        while get_elapsed() < MAX_CLUSTER_START_TIME:
+            status_response = self.get_cluster_status()
+            if status_response.get("state") == "Running":
+                return
+            else:
+                time.sleep(5)
+
+        raise dbt.exceptions.DbtRuntimeError(f"Cluster {self.cluster_id} restart timed out")
```

Comment:
maybe say after how long?

Reply:
Fixed in 8d00df96915e42ffa5a0f7c2782f35ca332f8faf